Sources/RingPublishingTracking/Public/Model/TrackingIdentifierError.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  TrackingIdentifierError.swift
+//  RingPublishingTracking
+//
+//  Created by Adam Szeremeta on 14/01/2022.
+//  Copyright © 2022 Ringier Axel Springer Tech. All rights reserved.
+//
+
+import Foundation
+
+/// TrackingIdentifierError
+public enum TrackingIdentifierError: Error {
+
+    /// Debug / opt-out mode is enabled and tracking identifier will not be retrieved
+    case debugModeEnabled
+
+    /// Url constructed to fetch tracking identifier was invalid
+    case invalidUrl
+
+    /// Tracking identifier request body could not be parsed
+    case incorrectRequestBody
```

Comment:
hmm.. nie lubię tego, że publicznie wystawiamy konkret o błędzie. co to obchodzi klienta?

Reply:
idąc tropem spójności w GDPR też wystawiliśmy to publicznie, oni lubią takie rzeczy :P
Zawsze to jakieś info co tam się nie udało a nie że nie udało i koniec kropka.

Reply:
tu trzeba naprawdę ostrożnie. Bo za każdy taki konkretny błąd będziemy rozliczani.

Reply:
Dla mnie @adamszeremeta w GDPR to zupełnie inna klasa błędów. Nie będąca wynikiem błędów po stronie modułu, a zdarzeń rzeczywiście możliwych.

https://github.com/ringpublishing/RingPublishingGDPR-iOS/blob/master/RingPublishingGDPR/Classes/Public/RingPublishingGDPRError.swift